src/components/Sign/Sign.tsx

```diff
@@ -1,118 +1,276 @@
-    ]
+import { ModalSignWrapper } from './Sign.styles';
+import {
+  fetchGeckoPrices,
+} from './Values/fiatValueManager';
+import { formatNearAmount } from './Values/formatNearAmount';
+import fiatValuesStore from './Values/store';
+import ArrowDownSvg from '../../Images/arrow-down';
+import ArrowUpSvg from '../../Images/arrow-up';
+import InternetSvg from '../../Images/Internet';
+import { Button } from '../../lib/Button';
+import { useAuthState } from '../../lib/useAuthState';
+import { network } from '../../utils/config';
+import TableContent from '../TableContent/TableContent';
+
+const formatActionType = (action: string) => {
+  switch (action) {
+    case 'createAccount':
+      return 'create_account';
+    case 'deployContract':
+      return 'deploy_contract';
+    case 'functionCall':
+      return 'function_call';
+    case 'addKey':
+      return 'add_key';
+    case 'deleteKey':
+      return 'delete_key';
+    case 'deleteAccount':
+      return 'delete_account';
+    case 'signedDelegate':
+      return 'signed_delegate';
+    default:
+      return action;
   }
-];
-*/
+};
+
+const deserializeTransactionsFromString = (transactionsString: string) => transactionsString
+  .split(',')
+  .map((str) => Buffer.from(str, 'base64'))
+  .map((buffer) => utils.serialize.deserialize(
+    transaction.SCHEMA,
+    transaction.Transaction,
+    buffer
+  ));
+
+interface TransactionDetails {
+  signerId: string;
+  receiverId: string;
+  totalAmount: string;
+  fees: {
+    transactionFees: string;
+    gasLimit: string;
+    gasPrice: string;
+  };
+  transactions: transaction.Transaction[];
+  actions: transaction.Action[];
+}
+
+export const calculateGasLimit = (actions) => actions
+  .filter((a) => Object.keys(a)[0] === 'functionCall')
+  .map((a) => a.functionCall.gas)
+  .reduce((totalGas, gas) => totalGas.add(gas), new BN(0))
+  .toString();
+
 function Sign() {
+  const [searchParams] = useSearchParams();
+  const [callbackUrl] = React.useState(searchParams.get('callbackUrl'));
+  const [transactionDetails, setTransactionDetails] =    React.useState<TransactionDetails>({
+    signerId:    '',
+    receiverId:  '',
+    totalAmount: '0',
+    fees:        {
+      transactionFees: '',
+      gasLimit:        '',
+      gasPrice:        '',
+    },
+    transactions: [],
+    actions:      [],
+  });
+  const authenticated = useAuthState();
   const [showDetails, setShowDetails] = React.useState(false);
+
+  const storeFetchedUsdValues = fiatValuesStore(
+    (state) => state.storeFetchedUsdValues
+  );
+
+  React.useEffect(() => {
+    if (authenticated === false) {
+      const parsedUrl = new URL(`${window.location.origin}/fastauth/login`);
+      parsedUrl.searchParams.set('success_url', window.location.href);
+      window.location.replace(parsedUrl.href);
+    }
+
+    const transactionHashes = searchParams.get('transactions');
+    const deserializedTransactions =      deserializeTransactionsFromString(transactionHashes);
```

Comment:
Can we redirect to `failure_url` if invalid borsh is passed in? Please address in a follow up PR